lib/logstash/filters/clone.rb

```diff
@@ -2,33 +2,86 @@
 require "logstash/filters/base"
 require "logstash/namespace"
 
-# The clone filter is for duplicating events.
-# A clone will be made for each type in the clone list.
-# The original event is left unchanged.
+# The clone filter is for creating copy of the original event. 
+# *The original event is left unchanged,* all modifier such as `add_field` or `add_tag`
+# will be applied to the newly created copy.
+#
+# The default configuration will create an exact copy of the event
+# [source, ruby]
+# filter {
+#   clone {}
+# }
+#
+# You can manipulate the copy using the `add_field`, `add_tag` configurations.
+# Following configuration will add a `clone` tag to the copy event.
+# [source, ruby]
+# filter {
+#   clone {
+#     add_tag => "clone"
+#   }
+# }
+#
+# In case you need to create several copies of the original event, you must set 
+# `mode => "multiple"` and specifies an identifier for each copy. 
+# By default this identifier is stored in the `type` field, 
+# but you can use a different field specifying the `clone_field` setting. 
+#
+# For example, following configuration will create 3 copies of the original event
+# [source,ruby]
+# filter {
+#   clone {
+#     mode => "multiple"
+#     clones => ["clone1","clone2","clone3"]
+#   }
+# }
+#
 class LogStash::Filters::Clone < LogStash::Filters::Base
 
   config_name "clone"
-
-  # A new clone will be created with the given type for each type in this list.
+  
+  # How this filter should operate
+  # * 'single' => create an exact copy
+  # * 'multiple' => create several copies using identifiers from the `clones` config
+  config :mode, :validate => ["single", "multiple"], :default => "single"
```

Comment:
I've been thinking about this, what if instead of having single/multiple setting with :mode, set instead a default for :clones ? For example:

` config :clones, :validate => :array, :default => ["clone"]`

this way the default behaviour of `clone {}` is the creation of 1 clone, and the clone will have `type => clone`